src/Controller/Api/PaymentMethodController.php

```
@@ -0,0 +1,51 @@
+<?php
+
+namespace Kiener\MolliePayments\Controller\Api;
+
+use Exception;
+use Kiener\MolliePayments\Service\PaymentMethodService;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\Routing\Annotation\RouteScope;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\Routing\Annotation\Route;
+
+class PaymentMethodController extends AbstractController
+{
+    /**
+     * @var PaymentMethodService
+     */
+    private $paymentMethodService;
+
+
+    /**
+     * @param PaymentMethodService $paymentMethodService
+     */
+    public function __construct(PaymentMethodService $paymentMethodService)
+    {
+        $this->paymentMethodService = $paymentMethodService;
+    }
+
+    /**
+     * @RouteScope(scopes={"api"})
+     * @Route("/api/_action/mollie/payment-method/update-methods",
+     *         defaults={"auth_enabled"=true}, name="api.mollie.payment-method.update-methods", methods={"GET"})
+     *
+     * @param Context $context
+     * @return JsonResponse
+     */
+    public function updatePaymentMethods(Context $context): JsonResponse
+    {
+        $success = true;
+
+        try {
+            $this->paymentMethodService->installAndActivatePaymentMethods($context);
+        } catch (Exception $e) {
+            $success = false;
```

i think we really want a logger in here?

also, what happens if you then throw the exception again? does it also lead to an error in the admin?
then we can make it fail fast
would it expose confidental data? then keep it as it is,

---

I'm not sure. It could both lead to database errors of Shopware, as well as API exceptions from Mollie. I don't exactly know which errors could be thrown and if there might be confidential data in there.

On the other hand, the user has access to the plugin configuration if it triggers this method. That wouldn't be someone who shouldn't have access to the data, right?

---

Correction: it could not lead to API exceptions from Mollie. It doesn't do anything with Mollie. It just works with the handler classes. So it would only be database errors of Shopware. I guess those are safe to show, right?